patching/apply-patch/src/index.ts

```diff
@@ -25,6 +29,13 @@ export function applyPatchToDir (opts: ApplyPatchToDirOpts): void {
     process.chdir(cwd)
   }
   if (!success) {
-    throw new PnpmError('PATCH_FAILED', `Could not apply patch ${opts.patchFilePath} to ${opts.patchedDir}`)
+    const message = `Could not apply patch ${opts.patchFilePath} to ${opts.patchedDir}`
+    if (opts.allowFailure) {
+      globalWarn(message)
+      return 'skipped'
+    } else {
+      throw new PnpmError('PATCH_FAILED', message)
+    }
   }
+  return 'applied'
```

Comment:
I think a boolean is enough
```suggestion
  return success
```

Reply:
https://github.com/pnpm/pnpm/pull/8337/commits/6cd2f20ca2a3d6b6fda8cd1ca82313306e9727b3